ydb/public/sdk/cpp/client/ydb_query/client.h

```diff
@@ -58,10 +59,16 @@ class TSession;
 class TQueryClient {
     friend class TSession;
     friend class NRetry::Async::TRetryContext<TQueryClient, TAsyncExecuteQueryResult>;
+    friend class NRetry::Async::TRetryContext<TQueryClient, TAsyncStatus>;
+    friend class NRetry::Sync::TRetryContext<TQueryClient, TStatus>;
 
 public:
-    using TQueryFunc = std::function<TAsyncExecuteQueryResult(TSession session)>;
+    using TQueryResultFunc = std::function<TAsyncExecuteQueryResult(TSession session)>;
+    using TQueryStatusFunc = std::function<TAsyncStatus(TSession session)>;
```

Comment:
```suggestion
    using TQueryFunc = std::function<TAsyncStatus(TSession session)>;
```
Result мы все равно уберем потом, так что просто на старое название поставим версию со статусом